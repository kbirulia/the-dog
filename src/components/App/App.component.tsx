import React from "react";
import { Switch, Route } from "react-router-dom";
import Header from "../Header";
import AllBreeds from "../AllBreeds";
import ErrorBoundary from "../ErrorBoundary";

const App = (): JSX.Element => (
  <ErrorBoundary>
    <Header />
    <main className="main">
      <Switch>
        <Route path="/" exact component={AllBreeds} />
        <Route
          path="/:breed_id"
          exact
          component={() => <div>Some breed</div>}
        />
      </Switch>
    </main>
  </ErrorBoundary>
);

export default App;
