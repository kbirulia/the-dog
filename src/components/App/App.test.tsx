import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import App from "./App.component";

jest.mock("../AllBreeds/AllBreeds.component");

describe("App", () => {
  test("should render all breeds", () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={["/"]} initialIndex={0}>
          <App />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
