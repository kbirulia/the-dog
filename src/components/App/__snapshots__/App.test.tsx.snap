// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render all breeds 1`] = `
<h1>
  Sorry.. there was an error
</h1>
`;
